queries/snakemake/highlights.scm

```diff
@@ -0,0 +1,63 @@
+; inherits: python
+
+;; Compound directives
+["rule" "checkpoint" "module"] @keyword
+
+;; Simple directives
+(directive name: _ @keyword)
+
+
+;; Rule imports
+(rule_import
+  "use" @include
+  "rule" @include
+  "from" @include
+  "as"? @include
+  "with"? @include
+)
+
+
+;; Rule inheritance
+(rule_inheritance
+  "use" @keyword
+  "rule" @keyword
+  "with"? @keyword
+)
```

Comment:
```suggestion
[
  "use"
  "rule"
  "with"
] @include
```

Reply:
Thank you for your comments! These are, however, not equivalent. The rule import imports a rule from an external file, the rule inheritance modifies a rule defined in the current file with alternative parameters. 

![1](https://user-images.githubusercontent.com/8322329/237026879-a8b29a8d-9260-4c36-ac08-82a5687c4107.jpg)

But actually, the `?` in `"with"?` for inheritance can go.

```suggestion
;; Rule imports
(rule_import
  "use" @include
  "rule" @include
  "from" @include
  "as"? @include
  "with"? @include
)


;; Rule inheritance
(rule_inheritance
  "use" @keyword
  "rule" @keyword
  "with" @keyword
)
```

Reply:
I would argue they almost do the same thing, you're using something defined somewhere else, one just changed parameters, I don't think its relative location is a candidate for having it marked as `@keyword` vs `@include`, however the actual "rule" keyword itself could be `@keyword` vs use, with, as, from

Reply:
My argument here is that rule inheritance `use rule with` is basically equivalent to class inheritance, whereas `use rule from` is an import, they just happen to share the same syntax.
Do you still think it should be changed? `as` and `with` would still need to be scoped to avoid clashes with Python's context manager `with x as y`

Reply:
From snakemake's loose ebnf-like syntax spec: 
```
userule = "use" "rule" (identifier | "*") "from" identifier ["as" identifier] ["with" ":" norunparams]
```

Scope them, sure, as to avoid conflict w/ regular Python keywords but these should absolutely be consistent then, and I'd go so far as to say having them be two separate rules in the grammar is incorrect but that's out of the scope of this issue :)

Reply:
Naturally, I disagree :)
The snippet you show doesn't mark the `"from" identifier` part as optional, so it appears to refer to the import, not inheritance, and conceptually, I think they differ.

From a highlighting perspective, if I do

```python
class A:
    def greet(self):
        print("Hi")
    def leave(self):
        print("Bye")

class B(A):
    def greet(self):
        print("Hello")
```

then I don't expect the inheritance to be highlighted as an import. Rule inheritance is the same thing.

If I failed to persuade you now, I'll concede. Just wanted to make my reasoning clear :)

Reply:
I missed the from part being optional....now I can agree with you here.